src/txkube/test/test_authentication.py

```diff
@@ -184,6 +234,21 @@ def test_https_bearer_token_authorization(self):
         )
 
 
+    def test_hostname_does_not_resolve(self):
+        """
+        Specifying a hostname which cannot be resolved to an
+        IP address will result in an ``DNSLookupError``.
+        """
+        d = self._authorized_request(token="test", headers=Headers({}),
+                                     kubernetesHost=b"doesnotresolve")
+        result = SynchronousTestCase().failureResultOf(d, DNSLookupError)
```

Comment:
Oops. :(  I forgot that txkube tests don't use trial's `TestCase`.  I don't think instantiating `SynchronousTestCase` to call methods on it is a great idea.  It probably won't break but normal usage of is not for application code to instantiate this kind of thing.

So, instead, what I should have asked for in my previous review was use of `testtools.twistedsupport.failed`.  I think it'll look something like this:

    has_lookup_failure = failed(
        AfterPreprocessing(lambda f: str(f.value),
        Equals("DNS lookup failed: ..."),
    )
    self.assertThat(some_deferred, has_lookup_failure)

Sorry about  steering you wrong with `failureResultOf`.

Reply:
OK